vLLM Sleep mode blog

tanpinsiang opened the pull request:
vLLM Sleep Mode feature blog. 
Coverage
Benchmarks: 0.6B-235B params, A4000 and A100 GPUs, TP=1-4
Ablation studies: warmup impact, FP8 quantization
Interactive Plotly charts with full methodology

Solves the multi-model serving problem: models too large to fit simultaneously on GPU, but traditional reloading too slow for production. Sleep Mode makes multi-model switching fast.

tanpinsiang committed:
sleep mode blog first draft

Signed-off-by: PinSiang <pinsiang.tan@embeddedllm.com>

tanpinsiang committed:
minor edit.

Signed-off-by: PinSiang <pinsiang.tan@embeddedllm.com>

tanpinsiang committed:
edit cache warmup

Signed-off-by: PinSiang <pinsiang.tan@embeddedllm.com>

tanpinsiang committed:
added ack and image

Signed-off-by: PinSiang <pinsiang.tan@embeddedllm.com>

tjtanaa commented:
@youkaichao  @hmellor PTAL

tanpinsiang committed:
Minor changes

Signed-off-by: PinSiang <pinsiang.tan@embeddedllm.com>

hmellor commented on _posts/2025-10-26-sleep-mode.md:
````diff
@@ -0,0 +1,468 @@
+
+```bash
+# Terminal 1: Start Phi-3-vision
+export VLLM_SERVER_DEV_MODE=1
````
Is dev mode necessary? If yes, why?

tjtanaa replied:
Yes. Dev mode flag is needed as the sleep mode API are only exposed under development environment. They are not expose in inference production stack as users can break the deployment by resetting the weights and cache. The sleep mode endpoint is to be used in a closed secure environment like training or backend applications.

hmellor commented on _posts/2025-10-26-sleep-mode.md:
```diff
@@ -0,0 +1,468 @@
+
+The interactive chart below compares the performance of vLLM with and without Sleep Mode enabled. With Level 1 Sleep Mode, models can be put to sleep and woken up without the costly reload overhead, enabling efficient GPU sharing between models.
+
+<div style="margin: 2rem 0;">
+<script src="https://cdn.plot.ly/plotly-2.32.0.min.js"></script>
+<div id="plotly-sleep-mode" style="width: 100%; height: 250px;"></div>
+<div style="text-align:center; color:#666; font-size:0.85rem; margin-top:0.75rem;">
+  <strong>Model A:</strong> Qwen3-235B-A22B-Instruct-2507-FP8 (TP=4) | <strong>Model B:</strong> Qwen3-Coder-30B-A3B-Instruct (TP=1)<br>
+  GPU: A100 | vLLM 0.11.0 | Sleep Level: 1 | Compilation: <code style="font-size:0.8rem;">cudagraph_mode: FULL_AND_PIECEWISE</code><br>
+  
+</div>
+<script src="/assets/figures/2025-vllm-sleep-mode/plotly-sleep-mode.js"></script>
+</div>
```
In this plot, and the others like it, could you explain what we are comparing here?

It looks like we're comparing the time taken to do the following:

- Prompt A
- Prompt B
- Prompt A
- Prompt B
- Prompt A
- Prompt B

It'd be nice to make this really clear to readers

hmellor reviewed (approved):
This is a really nice blog (the interactive graphs are great!).

Just a couple of comments about minor things that it might be nice to address before publishing.

youkaichao commented on _posts/2025-10-26-sleep-mode.md:
please change the filename to `2025-10-26-sleep-mode.md` , the resulting url will be better.

youkaichao commented on assets/figures/2025-vllm-sleep-mode/sleepmode.png:
this previous figure looks better, can we use a similar image?

<img width="338" height="446" alt="Image" src="https://github.com/user-attachments/assets/ada0c0e4-2b1c-4775-b9dd-ec4ed536ed15" />

tjtanaa replied:
Glad that you really like this image 😆

youkaichao reviewed (approved):
the interactive plots look great, so amazing!

left 2 nit comments.

tanpinsiang committed:
Rename blog post and improve documentation clarity

- Rename blog post file to shorter name (sleep-mode.md)
- Clarify security warning about dev mode requirement
- Improve plot description to explain A→B→A→B switching pattern
- Update sleepmode.png image

Signed-off-by: PinSiang <pinsiang.tan@embeddedllm.com>

youkaichao commented on _posts/2025-10-26-sleep-mode.md:
```diff
@@ -0,0 +1,471 @@
+The future of LLM serving is multi-model. Sleep Mode makes it practical today.
+
+## Acknowledgements
+
```
```suggestion


```

youkaichao committed:
Apply suggestion from @youkaichao

youkaichao commented on _posts/2025-10-26-sleep-mode.md:
```diff
@@ -0,0 +1,472 @@
+
+## Acknowledgements
+
+
```
```suggestion
```

youkaichao committed:
Apply suggestion from @youkaichao

youkaichao committed:
Merge branch 'main' into sleep-mode
